playbooks/pre.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: Pre
+  hosts: all
+
+  roles:
+    - ensure-pip
+
+  tasks:
+    - name: Run install script
+      ansible.builtin.shell:
+        executable: /bin/bash
+        chdir: "{{ zuul.project.src_dir }}"
+        cmd: |
+          set -e
+          set -x
+
+          sudo pip3 install -r requirements.txt
```

Comment:
This won't work for bookworm and likely Ubuntu 24.04, will need to use a virtual environment. But this can be fixed in a followup.